lms/djangoapps/instructor/tests/test_enrollment.py

```diff
@@ -771,6 +837,9 @@ def create_user(self, course_id=None):
         email = "robot_no_user_exists_with_this_email@edx.org"
         if self.user:
             user = UserFactory()
```

Comment:
You should be able to change this to `UserFactory(is_active=is_active)` and not need the next few lines you added.